src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -2907,7 +2936,14 @@ private static CultureInfo GetCultureFromFormatProvider(IFormatProvider formatPr
             TypeConverter integerConverter = LanguagePrimitives.GetIntegerSystemConverter(resultType);
             try
             {
-                return integerConverter.ConvertFrom(strToConvert);
+                if (TryScanNumber(strToConvert, resultType, out object result))
+                {
+                    return result;
+                }
+                else
+                {
+                    return integerConverter.ConvertFrom(strToConvert);
```

Comment:
This line comes in two variants. One uses this `integerConverter` method, and the other uses `Convert.ChangeType()`. I'm not sure they're something I can just arbitrarily swap out like that.

The ParseException is in the method because the `integerConverter.ConvertFrom()` or `Convert.ChangeType()` path is the _fallback_ for each. If we caught the parse exception _above_ where this method is called, we can't fallback to it.